Cython/Compiler/ExprNodes.py

```diff
@@ -8866,6 +8866,10 @@ def coerce_to(self, dst_type, env):
                         if isinstance(value, CoerceToPyTypeNode):
                             value = value.arg
                         item.value = value.coerce_to(member.type, env)
+        elif dst_type.can_coerce_from_pyobject(env):
```

Comment:
```suggestion
        elif dst_type.is_cpp_class and dst_type.can_coerce_from_pyobject(env):
```

(make it a little more specific)